File: Client/src/app/models/order.ts
import * as moment from 'moment';

export class Order {
  id: number;
  orderNumber: number;
  date: moment.Moment;
  totalPrice: number;
  status: string;

  constructor(res: Order) {
    this.id = res.id;
    this.orderNumber = res.orderNumber;
    this.date = moment.utc(res.date);
    this.totalPrice = res.totalPrice;
    this.status = res.status;
  }
}

File: Server/Repositories/Models/Order.cs
using Server.Models;

namespace Server.Repositories.Models {
    public class Order {

        public int Id { get; set; }
        public long OrderNumber { get; set; }
        public DateTime Date { get; set; }
        public double TotalPrice { get; set; }
        public string Status { get; set; }
    }

    public class OrderItems {
        public int Id { get; set; }
        public IEnumerable<Product> Products { get; set; }
    }

}
